src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -118,35 +253,42 @@ private async Task<bool> IsEnabledAsync<TContext>(string feature, TContext appCo
 
                         var context = new FeatureFilterEvaluationContext()
                         {
-                            FeatureName = feature,
-                            Parameters = featureFilterConfiguration.Parameters 
+                            FeatureName = featureDefinition.Name,
+                            Parameters = featureFilterConfiguration.Parameters
                         };
 
                         //
-                        // IContextualFeatureFilter
-                        if (useAppContext)
+                        // IFeatureFilter
+                        if (filter is IFeatureFilter featureFilter)
                         {
-                            ContextualFeatureFilterEvaluator contextualFilter = GetContextualFeatureFilter(featureFilterConfiguration.Name, typeof(TContext));
-
-                            if (contextualFilter != null && await contextualFilter
-                                .EvaluateAsync(context, appContext, cancellationToken)
-                                .ConfigureAwait(false))
+                            if (await featureFilter.EvaluateAsync(context, cancellationToken).ConfigureAwait(false))
                             {
                                 enabled = true;
 
                                 break;
```

Comment:
NIT: the `break` is not necessary. If you like, the code can be simplified to

```cs
enabled = await featureFilter.EvaluateAsync(context, cancellationToken).ConfigureAwait(false)
```

Reply:
Can you help me out a bit with this one? Looks to me like the break is necessary to prevent unnecessary filter evaluation in the event that an iteration evaluates as true.

Reply:
Ah, you are right. Forgot we have a list of filters.